frontend/src/metabase/nav/components/AppBar/AppBarToggle.styled.tsx

```diff
@@ -1,13 +1,48 @@
 import styled from "@emotion/styled";
+import { css } from "@emotion/react";
 import { color } from "metabase/lib/colors";
 import Icon from "metabase/components/Icon";
+import { AppBarLeftContainer } from "./AppBarLarge.styled";
 
-export const SidebarButton = styled.button`
+interface SidebarButtonProps {
+  isSmallAppBar?: boolean;
+  isNavBarEnabled?: boolean;
+  isLogoVisible?: boolean;
+}
+
+export const SidebarButton = styled.button<SidebarButtonProps>`
   cursor: pointer;
   display: block;
+
+  ${({ isNavBarEnabled, isLogoVisible, isSmallAppBar }) =>
+    isLogoVisible && !isSmallAppBar
```

Comment:
We used to do nested style in `AppBarLarge`, so after moving logic to the component we need to inject `isSmallAppBar` so we know which app bar size we're rendering this component into.

Actually, it could be omitted since `AppBarLeftContainer` only exist for large app bar, but for clarity's sake. I think passing `isSmallAppBar` intentionally conveys the intent better.